ir/visitor.h

```diff
@@ -165,7 +165,10 @@ class Visitor {
     virtual bool join_flows(const IR::Node *) { return false; }
     void visit_children(const IR::Node *, std::function<void()> fn) { fn(); }
     class ChangeTracker;  // used by Modifier and Transform -- private to them
-#ifndef NDEBUG
+#ifdef NDEBUG
+#define override_unless_NDEBUG
+#else
+#define override_unless_NDEBUG  override
```

Comment:
How about the following pattern?
```
#ifdef NDEBUG
#define P4C_OVERRIDE_METHOD(...) __VA_ARGS__
#else
#define P4C_OVERRIDE_METHOD(...) __VA_ARGS__ = override
#endif

P4C_OVERRIDE_METHOD(void check_clone(const Visitor *));
```

And put the macro in a common utils.h file?